File: SEP_JMS.UI/src/interface/price.ts
import { JobType } from "../enums/jobType";

export interface PriceGroup {
  priceGroupId: number;
  name: string;
  description: string;
}

export interface PriceItem {
  priceGroupName: string;
  priceId: string;
  jobTypeId: JobType
  unitPrice: number;
  description: string;
}

File: SEP_JMS.API/SEP_JMS.Model/Models/Price.cs
using Microsoft.EntityFrameworkCore;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace SEP_JMS.Model.Models
{
    [Index(nameof(PriceGroupId), nameof(JobTypeId), IsUnique = true)]
    public class Price
    {
        [Key, DatabaseGenerated(DatabaseGeneratedOption.None)]
        public Guid PriceId { get; set; }

        public Guid JobTypeId { get; set; }

        public Guid PriceGroupId { get; set; }

        public int UnitPrice { get; set; }

        public string? Description { get; set; } = null;

        [DeleteBehavior(DeleteBehavior.Restrict)]
        public virtual JobType JobType { get; set; } = null!;

        [DeleteBehavior(DeleteBehavior.Restrict)]
        public virtual PriceGroup PriceGroup { get; set; } = null!;
    }
}
